tests/Feature/ScoutElasticSearchServiceProviderTest.php

```diff
@@ -57,6 +57,7 @@ public function test_config_with_cloud_id()
     {
         $this->app['config']->set('elasticsearch.cloud_id', 'Test:ZXUtY2VudHJhbC0xLmF3cy5jbG91ZC5lcy5pbyQ0ZGU0NmNlZDhkOGQ0NTk2OTZlNTQ0ZmU1ZjMyYjk5OSRlY2I0YTJlZmY0OTA0ZDliOTE5NzMzMmQwOWNjOTY5Ng==');
         $this->app['config']->set('elasticsearch.api_key', '123456');
+        $this->app['config']->set('elasticsearch.user', null);
```

Comment:
Could you ask why we added this line to the test? Did we get any errors here?

Reply:
Yes [chel my commit](https://github.com/matchish/laravel-scout-elasticsearch/pull/214/commits/9b6e091fc86d1e7a330bb5e927e2703885bb972e ) . Basically, elasticsearch 8 requires authentication to interact with it, so I had to set username and pass in docker-compose services to run tests...etc .... but the elasticsesarch also requires that you use either basicAuthentication or apiKey not both at the same time.

So when testing config with cloud id, you will get an error ("You can't use both Basic Authentication and Cloud Id at the same time"), so that added line will set username to null to replicate the real-life scenario").

Another solution for all of this would be to maybe mock the elasticsearch client? but I am not sure if this is possible or the right way to go!

Reply:
Ahh, I see what you mean. Thanks for the clarification.
For the mocking the client. I tried it before but as you know the client is a final class. An issue was created for creating an interface for client but I don't know it will be in short term. https://github.com/elastic/elasticsearch-php/issues/1227

Reply:
Yes, I have seen that issue, there is a PR that fixes that already. It will be great if we can mock the client and make testing way much easier.